### PR TITLE
upgrade ts to torchserve-nightly==2022.3.23.post in PT1.10

### PR DESCRIPTION
**Pull request by lxning**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/inference/docker/1.10/py3/Dockerfile.e3.cpu [hunk 1]
```diff
@@ -120,7 +120,7 @@ RUN pip install \
 
 RUN pip uninstall -y model-archiver multi-model-server \
  && pip install captum \
- && pip install torchserve==${TS_VERSION} \
+ && pip install torchserve-nightly==2022.3.23.post2 \
```
Comment (saimidu) [inline pytorch/inference/docker/1.10/py3/Dockerfile.e3.cpu:123]: This is great! Please also comment out the ARG for TS_VERSION on this dockerfile to prevent confusion at a later date.

File: pytorch/inference/docker/1.10/py3/Dockerfile.graviton.e3.cpu [hunk 2]
```diff
@@ -120,7 +120,7 @@ RUN pip install pillow==8.3.2 "awscli<2"
 
 RUN pip uninstall -y model-archiver multi-model-server \
  && pip install captum \
- && pip install torchserve==$TS_VERSION \
+ && pip install torchserve-nightly==2022.3.23.post2 \
```
Comment (saimidu) [inline pytorch/inference/docker/1.10/py3/Dockerfile.graviton.e3.cpu:123]: This is great! Please also comment out the ARG for TS_VERSION on this dockerfile to prevent confusion at a later date.

File: pytorch/inference/docker/1.10/py3/cu113/Dockerfile.e3.gpu [hunk 3]
```diff
@@ -166,7 +166,7 @@ RUN pip install \
 
 RUN pip uninstall -y model-archiver multi-model-server \
  && pip install captum \
- && pip install torchserve==${TS_VERSION} \
+ && pip install torchserve-nightly==2022.3.23.post2 \
```
Comment (saimidu) [inline pytorch/inference/docker/1.10/py3/cu113/Dockerfile.e3.gpu:169]: This is great! Please also comment out the ARG for TS_VERSION on this dockerfile to prevent confusion at a later date.

File: pytorch/inference/docker/1.10/py3/sdk1.17.1/Dockerfile.neuron [hunk 4]
```diff
@@ -107,7 +107,7 @@ RUN pip install --no-cache-dir -U \
 
 RUN pip install neuron-cc[tensorflow] --extra-index-url https://pip.repos.neuron.amazonaws.com \
  && pip install "torch-neuron>=1.10.1,<1.10.2" --extra-index-url https://pip.repos.neuron.amazonaws.com \
- && pip install torchserve==$TS_VERSION \
+ && pip install torchserve-nightly==2022.3.23.post2 \
```
Comment (saimidu) [inline pytorch/inference/docker/1.10/py3/sdk1.17.1/Dockerfile.neuron:110]: This is great! Please also comment out the ARG for TS_VERSION on this dockerfile to prevent confusion at a later date.

File: pytorch/inference/docker/1.11/py3/Dockerfile.cpu [hunk 5]
```diff
@@ -157,7 +157,7 @@ RUN pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pytho
 
 RUN pip uninstall -y model-archiver multi-model-server \
  && pip install captum \
- && pip install torchserve==${TS_VERSION} \
+ && pip install torchserve-nightly==2022.3.23.post2 \
```
Comment (saimidu) [inline pytorch/inference/docker/1.11/py3/Dockerfile.cpu:160]: Let's only make this change on PT 1.10 inference DLCs, for now, and make changes on other framework versions once this change goes GA.

File: pytorch/inference/docker/1.11/py3/cu115/Dockerfile.gpu [hunk 6]
```diff
@@ -208,7 +208,7 @@ RUN pip install --no-cache-dir -U \
 
 RUN pip uninstall -y model-archiver multi-model-server \
  && pip install captum \
- && pip install torchserve==${TS_VERSION} \
+ && pip install torchserve-nightly==2022.3.23.post2 \
```
Comment (saimidu) [inline pytorch/inference/docker/1.11/py3/cu115/Dockerfile.gpu:211]: Let's only make this change on PT 1.10 inference DLCs, for now, and make changes on other framework versions once this change goes GA.

**Comment by saimidu** [comment]
@lxning Please trigger PT 1.10 inference DLC builds by replacing the `pytorch/buildspec.yml` with the `pytorch/buildspec-1-10.yml` file, in order to build PT 1.10 DLCs instead of the PT 1.11 DLCs.

**Comment by lxning** [comment]
@saimidu pytorch/buildspec.yml also includes training. Could you pls confirm that the following actions are correct
1. mv pytorch/buildspec.yml pytorch/buildspec-1-11.yml
2. mv pytorch/buildspec-1-10-2.yml pytorch/buildspec.yml

**Comment by saimidu** [comment]
CPU and GPU inference DLCs pass all tests.

**Review by saimidu — APPROVED** [review]
